### ISSUE TITLE
Numba error running squidpy.gr.sepal

### ISSUE THREAD

**Issue by emdann** [theislab/squidpy#419]
Hi,
Trying to run the Sepal score example I get the following error (probably caused by `numba`?). Any idea what the problem might be?

## Minimal reproducible example

```python
import scanpy as sc
import squidpy as sq

adata1 = sq.datasets.visium_hne_adata()
sq.gr.spatial_neighbors(adata1)
genes = adata1.var_names[(adata1.var.n_cells > 100) & adata1.var.highly_variable][0:100]
sq.gr.sepal(adata1, max_neighs=6, genes=genes, n_jobs=1)
```

## Traceback

<details>

```pytb
---------------------------------------------------------------------------
TypingError                               Traceback (most recent call last)
<ipython-input-3-9675cc9d1a81> in <module>
----> 1 sq.gr.sepal(adata1, max_neighs=6, genes=genes, n_jobs=1)
      2 # adata.uns["sepal_score"].head(10)

~/my-conda-envs/emma_env/lib/python3.7/site-packages/squidpy/gr/_sepal.py in sepal(adata, max_neighs, genes, n_iter, dt, thresh, connectivity_key, spatial_key, layer, use_raw, copy, n_jobs, backend, show_progress_bar)
    120 
    121     # get saturated/unsaturated nodes
--> 122     sat, sat_idx, unsat, unsat_idx = _compute_idxs(g, spatial, max_neighs, "l1")
    123 
    124     # get counts

~/my-conda-envs/emma_env/lib/python3.7/site-packages/squidpy/gr/_sepal.py in _compute_idxs(g, spatial, sat_thresh, metric)
    297     sat, unsat = _get_sat_unsat_idx(g.indptr, g.shape[0], sat_thresh)
    298 
--> 299     sat_idx, nearest_sat, un_unsat = _get_nhood_idx(sat, unsat, g.indptr, g.indices, sat_thresh)
    300 
    301     # compute dist btwn remaining unsat and all sat

~/my-conda-envs/emma_env/lib/python3.7/site-packages/numba/core/dispatcher.py in _compile_for_args(self, *args, **kws)
    413                 e.patch_message(msg)
    414 
--> 415             error_rewrite(e, 'typing')
    416         except errors.UnsupportedError as e:
    417             # Something unsupported is present in the user code, add help info

~/my-conda-envs/emma_env/lib/python3.7/site-packages/numba/core/dispatcher.py in error_rewrite(e, issue_type)
    356                 raise e
    357             else:
--> 358                 reraise(type(e), e, None)
    359 
    360         argtypes = []

~/my-conda-envs/emma_env/lib/python3.7/site-packages/numba/core/utils.py in reraise(tp, value, tb)
     78         value = tp()
     79     if value.__traceback__ is not tb:
---> 80         raise value.with_traceback(tb)
     81     raise value
     82 

TypingError: Failed in nopython mode pipeline (step: nopython frontend)
No implementation of function Function(<built-in function contains>) found for signature:
 
 >>> contains(array(int64, 1d, C), int32)
 
There are 18 candidate implementations:
  - Of which 18 did not match due to:
  Overload of function 'contains': File: <numerous>: Line N/A.
    With argument(s): '(array(int64, 1d, C), int32)':
   No match.

During: typing of intrinsic-call at /home/jovyan/my-conda-envs/emma_env/lib/python3.7/site-packages/squidpy/gr/_sepal.py (336)

File "../../../../../../home/jovyan/my-conda-envs/emma_env/lib/python3.7/site-packages/squidpy/gr/_sepal.py", line 336:
def _get_nhood_idx(
    <source elided>
        for u in unsat_neigh:
            if u in sat:  # take the first saturated nhood
            ^
```

</details>

## Version

`squidpy=1.1.2`
`numba=0.51.2`

**Comment by michalk8** [comment]
Hi @emdann , seems like in `if u in unsat_neigh`, `u` is of type `int32`, whereas `unsa_neigh` is an `int64[]` array, though am not sure why it can't find an impl. that would upcast the `int32` to `int64`.
Upgrading to `numba>=0.52.0` seems to fix this issue, (the example above works), will update requirements.

**Comment by michalk8** [comment]
closed via #420
